server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/projection/cassandra/model/FirstEnqueuedItem.java

```diff
@@ -0,0 +1,39 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.projection.cassandra.model;
+
+public class FirstEnqueuedItem {
+
+    private final String queueName;
+    private final Long timeRangeStart;
```

Comment:
the commit is old, I removed this class

Reply:
But why should I use primitive type over wrapper class? E.g. In configuration classes, If I let my `private Integer bucketCount` be primitive type, then after the initialization, my `bucketCount` will be 0 instead of null and could be un expected value the caller might want

Reply:
you don't really want null values.
You may want Optional<Long> or long but not Long.

Reply:
If my properties are mandatory, then could I use `private final Long myField;` and make `Preconditions` null check in the constructor?

Reply:
What does it bring you that `long` doesn't ?
using long reduce the number of checks you need.